pipeline/jobs/gradle-test.yaml

```diff
@@ -0,0 +1,36 @@
+parameters:
+  - name: displayName
+    type: string
+    default: ''
+
+  - name: dependsOn
+    type: object
+    default: []
+
+  - name: subscriptionKey
+    type: string
+    default: ''
+
+  - name: apimHost
+    type: string
+    default: ''
+
+  - name: command
+    type: string
+    default: ''
+
+jobs:
+- job: ${{ parameters.displayName }}
+  dependsOn:
+  - ${{ if parameters.dependsOn }}:
+    - ${{ parameters.dependsOn }}
+  displayName: ${{ parameters.displayName }}
+  pool:
+    vmImage: 'ubuntu-latest'
+    
+  variables:
+    subscriptionKey: $[ dependencies.ExportKeyVault.outputs['exportKeyVaultSecret.hmi-apim-sub-key'] ]
```

Comment:
this needs to be corrected. this needs to be dynamically built from params...create a variable object and use ${{ insert }} :